metagraph/src/annotation/taxonomy/tax_classifier.cpp

```diff
@@ -0,0 +1,294 @@
+        }
+        std::vector<std::string> parts = utils::split_string(line, "\t");
+        if (parts.size() <= 2) {
+            logger->error("The Taxonomic tree filepath contains incomplete lines. "
+                          "Please make sure that this file was not manually modified: {}",
+                          tax_tree_filepath);
+            exit(1);
+        }
+        uint32_t act = std::stoul(parts[0]);
+        uint32_t parent = std::stoul(parts[2]);
+        full_parents_list[act] = parent;
+        node_parent[act] = parent;
+    }
+
+    std::vector<TaxId> relevant_taxids;
+    // 'considered_relevant_taxids' is used to make sure that there are no duplications in 'relevant_taxids'.
+    tsl::hopscotch_set<TaxId> considered_relevant_taxids;
+
+    if (accversion_to_taxid_map.size()) {
+        // Store only the taxonomic nodes that exists in the annotation matrix.
+        for (const pair<std::string, TaxId> &it : accversion_to_taxid_map) {
+            relevant_taxids.push_back(it.second);
+            considered_relevant_taxids.insert(it.second);
+        }
+    } else {
+        // If 'this->accversion_to_taxid_map' is empty, store the entire taxonomic tree.
+        for (auto it : full_parents_list) {
+            relevant_taxids.push_back(it.first);
+            considered_relevant_taxids.insert(it.first);
+        }
+    }
+    assert(relevant_taxids.size());
+
+    uint64_t num_taxid_failed = 0; // num_taxid_failed is used for logging only.
+    for (uint32_t i = 0; i < relevant_taxids.size(); ++i) {
+        const TaxId taxid = relevant_taxids[i];
+        if (!full_parents_list.count(taxid)) {
+            num_taxid_failed += 1;
+            continue;
+        }
+
+        if (not considered_relevant_taxids.count(full_parents_list[taxid])) {
+            relevant_taxids.push_back(full_parents_list[taxid]);
+            considered_relevant_taxids.insert(full_parents_list[taxid]);
+        }
+
+        // Check if the current taxid is the root.
+        if (taxid == full_parents_list[taxid]) {
+            root_node = taxid;
+        }
+    }
+    if (num_taxid_failed) {
+        logger->warn("During the tax_tree_filepath {} parsing, {} taxids were not found out of {} total evaluations.",
+                     tax_tree_filepath, num_taxid_failed, relevant_taxids.size());
+    }
+
+    // Construct the output tree.
+    for (const TaxId &taxid : relevant_taxids) {
+        if (taxid == root_node) {
+            continue;
+        }
+        (*tree)[full_parents_list[taxid]].push_back(taxid);
+    }
+}
+
+void TaxonomyClsAnno::dfs_statistics(const TaxId node,
+                                     const ChildrenList &tree,
+                                     std::vector<TaxId> *tree_linearization) {
+    node_to_linearization_idx[node] = tree_linearization->size();
+    tree_linearization->push_back(node);
+    uint32_t depth = 0;
+    if (tree.count(node)) {
+        for (const TaxId &child : tree.at(node)) {
```

Comment:
Double lookup (`count` + `at`), which could be avoided with `tree.find(node)`

Reply:
`_anno_matrix` will be used by `assign_class()` method in the next PR. 

Replaced the count+at with an `it = tree.find(node)`

Thanks for the review!